src/ssl_stubs.c

```diff
@@ -1377,15 +1377,12 @@ CAMLprim value ocaml_ssl_embed_socket(value socket_, value context)
 
   if (socket < 0)
     caml_raise_constant(*caml_named_value("ssl_exn_invalid_socket"));
-  caml_release_runtime_system();
```

Comment:
I hadn't noticed this before. Do we need a new embed socket function? We should preserve the old behavior for `ocaml_ssl_embed_socket`.

Reply:
I first duplicated this function, but looking at the documentation, I found that SSL_new and SSL_set_fd can not block and do not do any communication, so one function not acquiring the runtime is enough (or one acquiring it) 

But you are right, a lot of function like ocaml_ssl_get_file_descr, should probably not release the runtime lock neither. But maybe you want this in another PR and we currently keep the previous one.

I just reverted (commit [606c1e5](https://github.com/savonet/ocaml-ssl/pull/106/commits/606c1e58286c6c94ce851e7206b725b710ddec2a)) and will submit an issue now.

Reply:
I just openned a new issue #113, if you could tell me what you think, I might propose a PR very quickly (or not at all if you don't like any of the two ideas ;-) 

More generally, there is a lot of unanswered issues on ocaml-ssl, this is not a very good for the image of the project. May be some of them desserve to be closed, other to have at least a short answer. I could actually answer some of them maybe ?

Reply:
Good point. Some issue gardening is probably in order. Feel free to answer them, of course.

I like your idea for the latest one, I'll answer there.

Reply:
I will first fix issue #103 (and even output_char), I think this is a serious bug (but probably no one is using them ?). I might look at all the other occurrences of ``ignore`` while I am at it.

Many thanks again for your review.

Reply:
I commented a few issues (only one has no answer now) and probably #31 and #42 can be closed.